Source/WebKit/UIProcess/API/Cocoa/_WKRemoteWebInspectorViewController.mm

```diff
@@ -106,6 +106,7 @@ - (instancetype)initWithConfiguration:(_WKInspectorConfiguration *)configuration
 
 - (void)dealloc
 {
+    m_remoteInspectorProxy->setClient(nullptr);
```

Comment:
It's usually better to not tie this kind of semantic to object lifetimes since ARC can make them unpredictable or surprising. Maybe it is OK in this case. But it's a bit of an anti-pattern.